src/overseer/store/jdbc.clj

```diff
@@ -0,0 +1,238 @@
+    (let [num-rows-updated (first (j/update! db-spec :overseer_jobs set-map' where-clause))]
+      (if (not= 1 num-rows-updated)
+        (throw (ex-info (str "Attempted to update stale job " (pr-str job-id)) {:type :stale-object}))
+        num-rows-updated))))
+
+(defn- direct-dependents
+  "Find the set of all Job IDs that directly depend on given `job-id`"
+  [db-spec job-id]
+  (->> (j/query db-spec ["SELECT job_id FROM overseer_dependencies WHERE dep_id = ?" job-id])
+       (map :job_id)
+       set))
+
+(defn dependents [db-spec job-id]
+  "Returns the set of all Job IDs that depend upon given `job-id`, both
+  directly and transitively"
+  (loop [all-dependents #{}
+         visited #{}
+         to-visit #{job-id}]
+    (if (empty? to-visit)
+      (set (map #(:job/id (query-job db-spec %)) all-dependents))
+      (let [job-id' (first to-visit)
+            dependents (direct-dependents db-spec job-id')
+            all-dependents' (set/union all-dependents dependents)
+            visited' (conj visited job-id')
+            to-visit' (set/difference
+                        (set/union to-visit dependents)
+                        visited')]
+        (recur all-dependents' visited' to-visit')))))
+
+(defn job-dep-jdbc-maps
+  "Return a seq of dependency rows to insert for a Graph<Job>"
+  [graph]
+  (let [nodes (loom.graph/nodes graph)
+        edges (loom.graph/edges graph)
+
+        dep-edges ; Basically recreate adjacency list {job [dep-jobs]}
+        (std/map-kv (partial map second) (group-by first edges))
+
+        job->dep-maps
+        (fn [job]
+          (when (contains? dep-edges job)
+            (->> (get dep-edges job)
+                 (map (fn [dep]
+                        {:job_id (:job/id job)
+                         :dep_id (:job/id dep)})))))]
+    (->> nodes
+         (mapcat job->dep-maps)
+         (filter identity))))
+
+(def default-table-opts
+  "Backend-specific system configuration options"
+  {:mysql
+   {:table-spec "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC"}})
+
+(defn install' [adapter db-spec]
+  (j/db-do-commands
+    db-spec
+    [(j/create-table-ddl
+       :overseer_jobs
+       [[:id "VARCHAR(64) PRIMARY KEY"]
+        [:type "VARCHAR(255)"]
+        [:args "VARCHAR(2048)"]
+        [:status "VARCHAR(255)"]
+        [:failure "VARCHAR(2048)"]
+        [:heartbeat :int]
+        [:lock_version "INT NOT NULL DEFAULT '0'"]]
+       (get default-table-opts adapter {}))
+     (j/create-table-ddl
+       :overseer_dependencies
+       [[:job_id "VARCHAR(64)"]
+        [:dep_id "VARCHAR(64)"]]
+       (get default-table-opts adapter {}))
+     "CREATE INDEX index_overseer_jobs_on_status ON overseer_jobs (status)"
+     "CREATE INDEX index_overseer_dependencies_on_job_id ON overseer_dependencies (job_id)"
+     "CREATE INDEX index_overseer_dependencies_on_dep_id ON overseer_dependencies (dep_id)"])
+  :ok)
+
+(defn- jobs-with-unfinished-dep
```

Comment:
I don't think it should be necessary to do queries, pull both result sets to the client, and then set difference them locally. 
 Remember MySQL does not work at the same Datomic with regard to this!  The standard way of doing this is to do as much of the set operation on the MySQL box as possible, because it's generally fast and minimizes giant result sets being set over the network.  This pattern holds up until it overloads the single MySQL box itself, but I don't think we're going to do that.

(The set difference operation query should be doable all in one query in SQL, you can Google it.)

If you want to do a different kind of query pattern, that's possible, but I'd recommend the standard "do everything on the DB" by default unless you specifically are doing a very specialized query pattern that involves more local computation on the worker peers.  But if you wanted to do a specialized query pattern, then I wouldn't recommend this strategy (joining on status on the MySQL box).  If you wanted a specialized query pattern I'd think you'd do something very specific, like the job dependencies here should be immutable under normal circumstances and thus cacheable locally on the worker peer.  The only thing you need to fetch/update from the server is the current status of the job nodes, you shouldn't need to refetch/query the job edges themselves.